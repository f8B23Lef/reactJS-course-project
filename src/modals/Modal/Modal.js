import React from 'react';
import PropTypes from 'prop-types';
import { useKeyDown } from '../../hooks/index';
import { KEYS } from '../../utils/constants';
import './Modal.scss';

export default function Modal(props) {
  const {
    onClose,
    children,
  } = props;

  useKeyDown(KEYS.ESCAPE_KEY, onClose);

  return (
    <div className='modal'>
      <div className='modal-content'>
        {children}
      </div>
    </div>
  );
}

Modal.propTypes = {
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node,
};

Modal.defaultProps = {
  children: null,
};
