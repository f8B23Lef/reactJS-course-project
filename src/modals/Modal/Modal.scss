@import '../../styles/index.scss';

.modal {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 3;
}

.modal-content {
  padding: 20px;
  width: 80vh;
  border-radius: 4px;
  background-color: $dark-grey;
}
