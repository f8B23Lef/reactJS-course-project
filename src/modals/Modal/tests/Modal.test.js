/* eslint-disable react/no-children-prop */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import Modal from '../Modal';

describe('Modal', () => {
  const onCloseMock = jest.fn();
  const children = `
    <div>Children1</div>
    <div>Children2</div>
  `;

  it('renders Modal snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Modal
          children={children}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    expect(asFragment(<Modal
      children={children}
      onClose={onCloseMock}
    />)).toMatchSnapshot();
  });

  it('calls onClose() on Escape button click', () => {
    render(
      <BrowserRouter>
        <Modal
          children={children}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    fireEvent.keyDown(document.body, { key: 'Escape' });
    expect(onCloseMock).toHaveBeenCalled();
  });
});
