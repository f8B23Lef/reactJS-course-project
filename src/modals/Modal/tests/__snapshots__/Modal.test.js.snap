// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Modal snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      
    &lt;div&gt;Children1&lt;/div&gt;
    &lt;div&gt;Children2&lt;/div&gt;
  
    </div>
  </div>
</DocumentFragment>
`;
